File: ccif/src/java/other/org/himalay/ccif/NewFlowConfirm1Type1.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:12 EST 2016")

public  class NewFlowConfirm1Type1 extends   APDUBaseFactory.APDUBase { //Concrete type is NewFlowConfirm1Type1


// member static classes
    public  static  class StatusfieldCheck extends BinStruct implements PublicBinMsg {


// member static classes
        public  static  class ServiceTypeCheck extends BinStruct implements PublicBinMsg {




            // members variables
            // ip_address
            public long ip_address ;

            // Parent of this object
            StatusfieldCheck parent;
            public ServiceTypeCheck (StatusfieldCheck parent) // throws Exception
            {
                this.parent= parent;
                init();
            }
            public StatusfieldCheck getParent()
            {
                return parent;
            }

            private void init()
            {
                // Initialize ip_address

            }

            public int readNoHeader(DataInputStream istream) throws IOException
            {

                return read(istream);
            }

            public int read(DataInputStream istream) throws IOException
            {
                preRead();
                int retVal= 0;

                // read ip_address
                {
                    ip_address=(long)(BinPrimitive.readUI32(istream) );
                    retVal+=4;
                }

                postRead();
                return retVal;
            }


            public int write(DataOutputStream ostream) throws IOException
            {
                preWrite();
                int retVal= 0;



                // write ip_address
                BinPrimitive.writeUI32(ostream,ip_address);
                retVal +=4;
                postWrite();
                return retVal;
            }

            public int dump(DumpContext dc) throws IOException
            {
                dc.indent();
                dc.getPs().print("ServiceTypeCheck\n");
                dc.increaseIndent();
                int retVal= 0;
                // write ip_address
                dc.indent();
                dc.getPs().println("ip_address="+ip_address+"(0x"+ Long.toHexString(ip_address)+")") ;
                dc.decreaseIndent();
                return retVal;
            }


            // Getter for ip_address
            //public long getIp_address()
            //{
            //    return ip_address ;
            //}


            // Setter for ip_address
            //public void setIp_address(long val)
            //{
            //    this.ip_address= val;
            //}


            public int getSize() throws IOException
            {
                DataOutputStream dos= new DataOutputStream(new NullStream());
                return this.write(dos);
            }



        }

// End of code


        // members variables
        // flowId
        public int flowId ;
        // serviceType
        public short serviceType ;
        // ServiceTypeCheckInstance
        public ServiceTypeCheck ServiceTypeCheckInstance ;

        // Parent of this object
        NewFlowConfirm1Type1 parent;
        public StatusfieldCheck (NewFlowConfirm1Type1 parent) // throws Exception
        {
            this.parent= parent;
            init();
        }
        public NewFlowConfirm1Type1 getParent()
        {
            return parent;
        }

        private void init()
        {
            // Initialize flowId

            // Initialize serviceType

            // Initialize ServiceTypeCheckInstance
            //Conditional, will be initialized during read
        }

        public int readNoHeader(DataInputStream istream) throws IOException
        {

            return read(istream);
        }

        public int read(DataInputStream istream) throws IOException
        {
            preRead();
            int retVal= 0;

            // read flowId
            {
                flowId= BinPrimitive.readUI24(istream);
                retVal += 3;
            }
            // read serviceType
            {
                serviceType=(short)(istream.readUnsignedByte());
                retVal+=1;
            }
            // read ServiceTypeCheckInstance
            if (serviceType==0x01) {
                ServiceTypeCheckInstance = new ServiceTypeCheck(this);
                retVal += ServiceTypeCheckInstance.read(istream);
            }

            postRead();
            return retVal;
        }


        public int write(DataOutputStream ostream) throws IOException
        {
            preWrite();
            int retVal= 0;



            {   /** fix dependent sizes for ServiceTypeCheckInstance **/
            }

            // write flowId
            {
                ostream.writeByte((flowId  & 0x00FF0000)>>16);
                ostream.writeShort((flowId & 0x0000FFFF));
                retVal+=3;
            }
            // write serviceType
            ostream.writeByte(serviceType);
            retVal +=1;
            // write ServiceTypeCheckInstance
            if (ServiceTypeCheckInstance!=null)retVal +=ServiceTypeCheckInstance.write(ostream);
            postWrite();
            return retVal;
        }

        public int dump(DumpContext dc) throws IOException
        {
            dc.indent();
            dc.getPs().print("StatusfieldCheck\n");
            dc.increaseIndent();
            int retVal= 0;
            // write flowId
            dc.indent();
            dc.getPs().println("flowId="+flowId+"(0x"+ Integer.toHexString(flowId)+")") ;
            // write serviceType
            dc.indent();
            dc.getPs().println("serviceType="+serviceType+"(0x"+ Integer.toHexString(serviceType) + ")" );
            // write ServiceTypeCheckInstance
            if ( ServiceTypeCheckInstance != null ) {
                dc.indent();
                dc.getPs().println("ServiceTypeCheckInstance") ;
                retVal +=ServiceTypeCheckInstance.dump(dc);
            }
            dc.decreaseIndent();
            return retVal;
        }


        // Getter for flowId
        //public int getFlowId()
        //{
        //    return flowId ;
        //}


        // Setter for flowId
        //public void setFlowId(int val)
        //{
        //    this.flowId= val;
        //}
        // Getter for serviceType
        //public short getServiceType()
        //{
        //    return serviceType ;
        //}


        // Setter for serviceType
        //public void setServiceType(short val)
        //{
        //    this.serviceType= val;
        //}
        // Getter for ServiceTypeCheckInstance
        //public ServiceTypeCheck getServiceTypeCheckInstance()
        //{
        //    return ServiceTypeCheckInstance ;
        //}


        // Setter for ServiceTypeCheckInstance
        //public void setServiceTypeCheckInstance(ServiceTypeCheck val)
        //{
        //    this.ServiceTypeCheckInstance= val;
        //}


        public int getSize() throws IOException
        {
            DataOutputStream dos= new DataOutputStream(new NullStream());
            return this.write(dos);
        }



    }

// End of code


    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // statusField
    public short statusField ;
    // flowsRemaining
    public short flowsRemaining ;
    // StatusfieldCheckInstance
    public StatusfieldCheck StatusfieldCheckInstance ;

    public NewFlowConfirm1Type1 () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length

        // Initialize statusField

        // Initialize flowsRemaining

        // Initialize StatusfieldCheckInstance
        //Conditional, will be initialized during read
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        preRead();
        int retVal= 0;
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read statusField
        {
            statusField=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read flowsRemaining
        {
            flowsRemaining=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read StatusfieldCheckInstance
        if (statusField == 0) {
            StatusfieldCheckInstance = new StatusfieldCheck(this);
            retVal += StatusfieldCheckInstance.read(istream);
        }

        postRead();
        return retVal;
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read header
        retVal += header.read(istream);
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read statusField
        {
            statusField=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read flowsRemaining
        {
            flowsRemaining=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read StatusfieldCheckInstance
        if (statusField == 0) {
            StatusfieldCheckInstance = new StatusfieldCheck(this);
            retVal += StatusfieldCheckInstance.read(istream);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for header **/
        }



        {   /** fix dependent sizes for StatusfieldCheckInstance **/
        }

        // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length);
        retVal +=1;
        // write statusField
        ostream.writeByte(statusField);
        retVal +=1;
        // write flowsRemaining
        ostream.writeByte(flowsRemaining);
        retVal +=1;
        // write StatusfieldCheckInstance
        if (StatusfieldCheckInstance!=null)retVal +=StatusfieldCheckInstance.write(ostream);
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("NewFlowConfirm1Type1\n");
        dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {
            dc.indent();
            dc.getPs().println("header") ;
            retVal +=header.dump(dc);
        }
        // write length
        dc.indent();
        dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write statusField
        dc.indent();
        dc.getPs().println("statusField="+statusField+"(0x"+ Integer.toHexString(statusField) + ")" );
        // write flowsRemaining
        dc.indent();
        dc.getPs().println("flowsRemaining="+flowsRemaining+"(0x"+ Integer.toHexString(flowsRemaining) + ")" );
        // write StatusfieldCheckInstance
        if ( StatusfieldCheckInstance != null ) {
            dc.indent();
            dc.getPs().println("StatusfieldCheckInstance") ;
            retVal +=StatusfieldCheckInstance.dump(dc);
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}


    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}


    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for statusField
    //public short getStatusField()
    //{
    //    return statusField ;
    //}


    // Setter for statusField
    //public void setStatusField(short val)
    //{
    //    this.statusField= val;
    //}
    // Getter for flowsRemaining
    //public short getFlowsRemaining()
    //{
    //    return flowsRemaining ;
    //}


    // Setter for flowsRemaining
    //public void setFlowsRemaining(short val)
    //{
    //    this.flowsRemaining= val;
    //}
    // Getter for StatusfieldCheckInstance
    //public StatusfieldCheck getStatusfieldCheckInstance()
    //{
    //    return StatusfieldCheckInstance ;
    //}


    // Setter for StatusfieldCheckInstance
    //public void setStatusfieldCheckInstance(StatusfieldCheck val)
    //{
    //    this.StatusfieldCheckInstance= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }


    public void setHeader(APDUHeader header)
    {
        this.header= header;
    }

    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/NewFlowConfirm1Type1.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:19 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "NewFlowConfirm1Type1.h"

#ifdef org_himalay_ccif_NewFlowConfirm1Type1__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;
// member static classes
// member static classes
// Constructor
NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck::ServiceTypeCheck() {
    references= 0;

    parent = NULL;

    // ip_address
    ip_address= (ui32)0;
}
// Destructor
NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck::~ServiceTypeCheck() {
    // ip_address

    parent = NULL;
}

int NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // ip_address
    {
        ip_address=istream.readUI32() ;
        retVal+=4;
    }
    return retVal;
}

int NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read ip_address
    {
        ip_address=istream.readUI32() ;
        retVal+=4;
    }

    return retVal;
}

int NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck::write(DataOutputStream& ostream) { // throws IOException


    int retVal= 0;
    // write ip_address
    ostream.writeUI32(ip_address);
    retVal +=4;

    return retVal;
}

int NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("ServiceTypeCheck\n");
    dc.increaseIndent();
    int retVal= 0;
    // write ip_address
    dc.indent();
    dc.getPs().print("ip_address=");
    dc.getPs().println((long)ip_address);
    dc.decreaseIndent();
    return retVal;
}


// Getter for ip_address
//ui32 NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck::getIp_address()
//{
//return this->ip_address;
//}

// Setter for ip_address
//void NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck::setIp_address(ui32 val)
//{
//this->ip_address= val;
//}

int NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck_ServiceTypeCheck__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck* p) {
    p->decreaseRef();
};;
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck* p) {
    p->decreaseRef();
};;
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck* p) {
    p->decreaseRef();
};;
#endif

// Constructor
NewFlowConfirm1Type1::StatusfieldCheck::StatusfieldCheck() {
    references= 0;

    parent = NULL;

    // flowId
    flowId= (ui24)0;
    // serviceType
    serviceType= (ui8)0;
    // ServiceTypeCheckInstance
    org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck_ServiceTypeCheck__INIT(ServiceTypeCheckInstance)
}
// Destructor
NewFlowConfirm1Type1::StatusfieldCheck::~StatusfieldCheck() {
    // flowId

    // serviceType

    // ServiceTypeCheckInstance
    org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck_ServiceTypeCheck__DELETE( ServiceTypeCheckInstance);
    parent = NULL;
}

int NewFlowConfirm1Type1::StatusfieldCheck::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // flowId
    {
        int i1= (0x000000FF & ((int)istream.readUI8())) << 16;
        int i2= (0x0000FFFF & ((int)istream.readUI16()));
        flowId= i1 | i2;
        retVal += 3;
    }
    // serviceType
    {
        serviceType=(istream.readUI8());
        retVal+=1;
    }
    // ServiceTypeCheckInstance
    if (serviceType==0x01) {
        org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck_ServiceTypeCheck__NEW( ServiceTypeCheckInstance);
        ServiceTypeCheckInstance->parent=this;
        retVal += ServiceTypeCheckInstance->read(istream);
    }
    return retVal;
}

int NewFlowConfirm1Type1::StatusfieldCheck::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read flowId
    {
        int i1= (0x000000FF & ((int)istream.readUI8())) << 16;
        int i2= (0x0000FFFF & ((int)istream.readUI16()));
        flowId= i1 | i2;
        retVal += 3;
    }
    // read serviceType
    {
        serviceType=(istream.readUI8());
        retVal+=1;
    }
    // read ServiceTypeCheckInstance
    if (serviceType==0x01) {
        org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck_ServiceTypeCheck__NEW( ServiceTypeCheckInstance);
        ServiceTypeCheckInstance->parent=this;
        retVal += ServiceTypeCheckInstance->read(istream);
    }

    return retVal;
}

int NewFlowConfirm1Type1::StatusfieldCheck::write(DataOutputStream& ostream) { // throws IOException


    {   /** fix dependent sizes for ServiceTypeCheckInstance **/
    }

    int retVal= 0;
    // write flowId
    {
        ostream.writeUI8((flowId  & 0x00FF0000)>>16);
        ostream.writeUI16((flowId & 0x0000FFFF));
        retVal+=3;
    }
    // write serviceType
    ostream.writeUI8(serviceType);
    retVal +=1;
    // write ServiceTypeCheckInstance
    if ( ServiceTypeCheckInstance != NULL ) {
        retVal +=ServiceTypeCheckInstance->write(ostream);
    }

    return retVal;
}

int NewFlowConfirm1Type1::StatusfieldCheck::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("StatusfieldCheck\n");
    dc.increaseIndent();
    int retVal= 0;
    // write flowId
    dc.indent();
    dc.getPs().print("flowId=");
    dc.getPs().println((long)flowId);
    // write serviceType
    dc.indent();
    dc.getPs().print("serviceType=");
    dc.getPs().println((long)serviceType);
    // write ServiceTypeCheckInstance
    if ( ServiceTypeCheckInstance != NULL ) {
        dc.indent();
        dc.getPs().println("ServiceTypeCheckInstance") ;
        retVal +=ServiceTypeCheckInstance->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for flowId
//ui24 NewFlowConfirm1Type1::StatusfieldCheck::getFlowId()
//{
//return this->flowId;
//}

// Setter for flowId
//void NewFlowConfirm1Type1::StatusfieldCheck::setFlowId(ui24 val)
//{
//this->flowId= val;
//}

// Getter for serviceType
//ui8 NewFlowConfirm1Type1::StatusfieldCheck::getServiceType()
//{
//return this->serviceType;
//}

// Setter for serviceType
//void NewFlowConfirm1Type1::StatusfieldCheck::setServiceType(ui8 val)
//{
//this->serviceType= val;
//}

// Getter for ServiceTypeCheckInstance
//org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck_ServiceTypeCheck__PTR_TYPE NewFlowConfirm1Type1::StatusfieldCheck::getServiceTypeCheckInstance()
//{
//return this->ServiceTypeCheckInstance;
//}

// Setter for ServiceTypeCheckInstance
//void NewFlowConfirm1Type1::StatusfieldCheck::setServiceTypeCheckInstance(org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck_ServiceTypeCheck__PTR_TYPE val)
//{
//this->ServiceTypeCheckInstance= val;
//}

int NewFlowConfirm1Type1::StatusfieldCheck::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck* p) {
    p->decreaseRef();
};;
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck* p) {
    p->decreaseRef();
};;
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck* p) {
    p->decreaseRef();
};;
#endif

// Constructor
NewFlowConfirm1Type1::NewFlowConfirm1Type1() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // statusField
    statusField= (ui8)0;
    // flowsRemaining
    flowsRemaining= (ui8)0;
    // StatusfieldCheckInstance
    org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck__INIT(StatusfieldCheckInstance)
}
// Destructor
NewFlowConfirm1Type1::~NewFlowConfirm1Type1() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // statusField

    // flowsRemaining

    // StatusfieldCheckInstance
    org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck__DELETE( StatusfieldCheckInstance);
}

int NewFlowConfirm1Type1::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // statusField
    {
        statusField=(istream.readUI8());
        retVal+=1;
    }
    // flowsRemaining
    {
        flowsRemaining=(istream.readUI8());
        retVal+=1;
    }
    // StatusfieldCheckInstance
    if (statusField == 0) {
        org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck__NEW( StatusfieldCheckInstance);
        StatusfieldCheckInstance->parent=this;
        retVal += StatusfieldCheckInstance->read(istream);
    }
    return retVal;
}

int NewFlowConfirm1Type1::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read statusField
    {
        statusField=(istream.readUI8());
        retVal+=1;
    }
    // read flowsRemaining
    {
        flowsRemaining=(istream.readUI8());
        retVal+=1;
    }
    // read StatusfieldCheckInstance
    if (statusField == 0) {
        org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck__NEW( StatusfieldCheckInstance);
        StatusfieldCheckInstance->parent=this;
        retVal += StatusfieldCheckInstance->read(istream);
    }

    return retVal;
}

int NewFlowConfirm1Type1::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }



    {   /** fix dependent sizes for StatusfieldCheckInstance **/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write statusField
    ostream.writeUI8(statusField);
    retVal +=1;
    // write flowsRemaining
    ostream.writeUI8(flowsRemaining);
    retVal +=1;
    // write StatusfieldCheckInstance
    if ( StatusfieldCheckInstance != NULL ) {
        retVal +=StatusfieldCheckInstance->write(ostream);
    }

    return retVal;
}

int NewFlowConfirm1Type1::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("NewFlowConfirm1Type1\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write statusField
    dc.indent();
    dc.getPs().print("statusField=");
    dc.getPs().println((long)statusField);
    // write flowsRemaining
    dc.indent();
    dc.getPs().print("flowsRemaining=");
    dc.getPs().println((long)flowsRemaining);
    // write StatusfieldCheckInstance
    if ( StatusfieldCheckInstance != NULL ) {
        dc.indent();
        dc.getPs().println("StatusfieldCheckInstance") ;
        retVal +=StatusfieldCheckInstance->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE NewFlowConfirm1Type1::getHeader()
//{
//return this->header;
//}

// Setter for header
//void NewFlowConfirm1Type1::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 NewFlowConfirm1Type1::getLength()
//{
//return this->length;
//}

// Setter for length
//void NewFlowConfirm1Type1::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for statusField
//ui8 NewFlowConfirm1Type1::getStatusField()
//{
//return this->statusField;
//}

// Setter for statusField
//void NewFlowConfirm1Type1::setStatusField(ui8 val)
//{
//this->statusField= val;
//}

// Getter for flowsRemaining
//ui8 NewFlowConfirm1Type1::getFlowsRemaining()
//{
//return this->flowsRemaining;
//}

// Setter for flowsRemaining
//void NewFlowConfirm1Type1::setFlowsRemaining(ui8 val)
//{
//this->flowsRemaining= val;
//}

// Getter for StatusfieldCheckInstance
//org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck__PTR_TYPE NewFlowConfirm1Type1::getStatusfieldCheckInstance()
//{
//return this->StatusfieldCheckInstance;
//}

// Setter for StatusfieldCheckInstance
//void NewFlowConfirm1Type1::setStatusfieldCheckInstance(org_himalay_ccif_NewFlowConfirm1Type1_StatusfieldCheck__PTR_TYPE val)
//{
//this->StatusfieldCheckInstance= val;
//}

int NewFlowConfirm1Type1::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void NewFlowConfirm1Type1::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE NewFlowConfirm1Type1::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_NewFlowConfirm1Type1__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck::ServiceTypeCheck* p) {
    p->decreaseRef();
};;
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowConfirm1Type1::StatusfieldCheck* p) {
    p->decreaseRef();
};;
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowConfirm1Type1* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowConfirm1Type1* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code